crates/ln-dlc-node/src/ldk_node_wallet.rs

```diff
@@ -205,6 +209,65 @@ where
         Ok(utxos)
     }
 
+    pub fn get_utxos_for_amount(
+        &self,
+        amount: u64,
+        lock_utxos: bool,
+        network: Network,
+    ) -> Result<Vec<Utxo>, Error> {
+        let utxos = self.get_utxos()?;
+        // get temporarily reserved utxo from in-memory storage
+        let mut reserved_outpoints = self.locked_outpoints.lock();
+
+        // filter reserved utxos from all known utxos to not accidentally double spend and those who
+        // have actually been spent already
+        let utxos = utxos
+            .iter()
+            .filter(|utxo| !reserved_outpoints.contains(&utxo.outpoint))
+            .filter(|utxo| !utxo.is_spent)
+            .collect::<Vec<_>>();
+
+        let mut utxos = utxos
+            .into_iter()
+            .map(|x| UtxoWrap {
+                utxo: Utxo {
+                    tx_out: x.txout.clone(),
+                    outpoint: x.outpoint,
+                    address: Address::from_script(&x.txout.script_pubkey, network)
+                        .expect("to be a valid address"),
+                    redeem_script: Default::default(),
+                    reserved: false,
+                },
+            })
+            .collect::<Vec<_>>();
+
+        // select enough utxos for our needs
+        let selected_local_utxos = select_coins(amount, 20, &mut utxos);
+        match selected_local_utxos {
+            None => Ok(vec![]),
+            Some(selected_local_utxos) => {
+                // update our temporarily reserved utxos with the selected once.
+                // note: this storage is only cleared up on a restart, meaning, if the protocol
+                // fails later on, the utxos will remain reserved
+                if lock_utxos {
+                    for utxo in selected_local_utxos.clone() {
+                        reserved_outpoints.push(utxo.utxo.outpoint);
+                    }
+                }
+
+                Ok(selected_local_utxos
+                    .into_iter()
+                    .map(|utxo| utxo.utxo)
+                    .collect())
+            }
+        }
+    }
+
+    pub fn add_reserved_outpoint(&self, outpoint: OutPoint) {
```

Comment:
🤔 it looks like this function isn't used anymore?

Reply:
good catch. Removed